scripts/update-references-in-vsce.js

```diff
@@ -0,0 +1,214 @@
+          if (err) {
+            console.error(`Error writing file ${filePath}: ${err}`);
+          } else {
+            console.log(`Updated dependencies in ${filePath}`);
+          }
+        });
+      }
+    });
+  }
+
+  // Function to update package.json directly under the packages directory
+  function traverseDirectories(directory) {
+    fs.readdir(directory, { withFileTypes: true }, (err, files) => {
+      if (err) {
+        console.error(`Error reading directory ${directory}: ${err}`);
+        return;
+      }
+
+      files.forEach(file => {
+        if (file.isDirectory()) {
+          const packageJsonPath = path.join(directory, file.name, 'package.json');
+          if (fs.existsSync(packageJsonPath)) {
+            updateDependencies(packageJsonPath);
+          }
+        }
+      });
+    });
+  }
+
+  // Start traversing from the packages directory
+  traverseDirectories(packagesDir);
+}
+
+function updateImports() {
+  const dirs = ['src', 'test'];
+  function replaceTextInFile(filePath) {
+    const data = fs.readFileSync(filePath, 'utf8');
+    let result = data.replace(
+      /'@salesforce\/core'/g,
+      "'@salesforce/core-bundle'"
+    );
+    result = result.replace(
+      /'@salesforce\/core\/(.+)'/g,
+      "'@salesforce/core-bundle'"
+    );
+    result = result.replace(
+      /'@salesforce\/source-deploy-retrieve/g,
+      "'@salesforce/source-deploy-retrieve-bundle"
+    );
+    result = result.replace(
+      /'@salesforce\/source-tracking/g,
+      "'@salesforce/source-tracking-bundle"
+    );
+    result = result.replace(
+      /'@salesforce\/templates/g,
+      "'@salesforce/templates-bundle"
+    );
+    result = result.replace(
+      /'@salesforce\/apex-node/g,
+      "'@salesforce/apex-node-bundle"
+    );
+
+    fs.writeFileSync(filePath, result, 'utf8');
+  }
+  function traverseDirectory(directory) {
+    fs.readdirSync(directory).forEach((file) => {
+      const fullPath = path.join(directory, file);
+      if (fs.lstatSync(fullPath).isDirectory()) {
+        traverseDirectory(fullPath);
+      } else if (path.extname(fullPath) === '.ts') {
+        replaceTextInFile(fullPath);
+      }
+    });
+  }
+
+  function traversePackages(directory) {
+    fs.readdir(directory, { withFileTypes: true }, (err, files) => {
+      if (err) {
+        console.error(`Error reading directory ${directory}: ${err}`);
+        return;
+      }
+
+      files.forEach(file => {
+        if (file.isDirectory()) {
+          dirs.forEach((dir) => {
+            const tsDirPath = path.join(directory, file.name, dir);
+            if (fs.existsSync(tsDirPath)) {
+              traverseDirectory(tsDirPath)
+            }
+          });
+        }
+      });
+    });
+  }
+  traversePackages(packagesDir);
+  console.log('imports updated successfully');
+}
+
+// Update the references in esbuild.config.js
+function updateEsbuildConfig() {
+  function replaceTextInFile(filePath) {
```

Comment:
This helper function is exactly the same as the one in Line 95.  Is it possible to avoid duplicate code by moving the helper function to the top level and calling that?